test/integration/network_integration_test.go

```diff
@@ -0,0 +1,307 @@
+// +build integration
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/cenkalti/backoff"
+	"github.com/cloudscale-ch/cloudscale-go-sdk"
+	"sync"
+	"testing"
+)
+
+const networkBaseName = "go-sdk-integration-test-network"
+
+func DeleteNetworkWithRetry(network *cloudscale.Network) error {
+	operation := func() error {
+		err := client.Networks.Delete(context.Background(), network.UUID)
+		if err != nil {
+			msg := fmt.Sprintf("Networks.Delete returned error %s\n", err)
```

Comment:
It might be worth limiting this to err to only `400 Bad Request` or just whatever is returned there. IMO it's quite bad that it just continues working in all kinds of cases. You can cast the error above to a cloudscale API error. I think that would make sense.

Reply:
We decided to leave this in-place until the internal ticket 2743 is resolved.